doc/sphinx/language/core/records.rst

```diff
@@ -1,248 +1,341 @@
+      is not zero.  :n:`Rat_irreducible` is a proof that the fraction is in
+      lowest terms.
 
-   Note here that the fields ``Rat_bottom_cond`` depends on the field ``bottom``
-   and ``Rat_irred_cond`` depends on both ``top`` and ``bottom``.
+.. _reuse_field_name:
 
-Let us now see the work done by the :cmd:`Record` command. First the command
-generates a variant type definition with just one constructor:
-:n:`Variant @ident {* @binder } : @sort := @ident__0 {* @binder }`.
+   .. example:: Reusing a field name in multiple records
 
-To build an object of type :token:`ident`, provide the constructor
-:n:`@ident__0` with the appropriate number of terms filling the fields of the record.
+      .. coqtop:: in
 
-.. example::
+         Module A. Record R := { f : nat }. End A.
+         Module B. Record S := { f : nat }. End B.
 
-   Let us define the rational :math:`1/2`:
+      .. coqtop:: all
 
-    .. coqtop:: in
+         Check {| A.f := 0 |}.
+         Check {| B.f := 0 |}.
 
-       Theorem one_two_irred : forall x y z:nat, x * y = 1 /\ x * z = 2 -> x = 1.
-       Admitted.
+.. _record_as_clause:
 
-       Definition half := mkRat true 1 2 (O_S 1) one_two_irred.
-       Check half.
+   .. example:: Using the "as" clause in a record definition
 
-Alternatively, the following syntax allows creating objects by using named fields, as
-shown in this grammar. The fields do not have to be in any particular order, nor do they have
-to be all present if the missing ones can be inferred or prompted for
-(see :ref:`programs`).
+      .. coqtop:: all
 
-.. coqtop:: all
+         Record MyRecord := { myfield : nat } as VarName.
+         About myfield. (* observe the MyRecord variable is named "VarName" *)
+         
+         (* make "VarName" implicit without having to rename the variable,
+            which would be necessary without the "as" clause *)
+         Arguments myfield {VarName}.   (* make "VarName" an implicit parameter *)
+         Check myfield.
+         Check (myfield (VarName:={| myfield := 0 |})).
 
-  Definition half' :=
-    {| sign := true;
-       Rat_bottom_cond := O_S 1;
-       Rat_irred_cond := one_two_irred |}.
+   .. exn:: Records declared with the keyword Record or Structure cannot be recursive.
 
-The following settings let you control the display format for types:
+      The record name :token:`ident` appears in the type of its fields, but uses
+      the :cmd:`Record` command. Use  the :cmd:`Inductive` or
+      :cmd:`CoInductive` command instead.
 
-.. flag:: Printing Records
+   .. exn:: @ident already exists
 
-   When this :term:`flag` is on (this is the default),
-   use the record syntax (shown above) as the default display format.
+      The fieldname :n:`@ident` is already defined as a global.
 
-You can override the display format for specified types by adding entries to these tables:
+   .. warn:: @ident__1 cannot be defined because the projection @ident__2 was not defined
+      :undocumented:
```

Comment:
Can you reintroduce the documentation for this warning, i.e.: ```The type of the projections :token:`ident` depends on previous projections which themselves could not be defined.```?

Reply:
Got it.  Updated.